code/asteroid/asteroid.cpp

```diff
@@ -796,35 +806,13 @@ void asteroid_create_debris_field(int num_asteroids, int asteroid_speed, int deb
 	Asteroid_field.speed = (float)asteroid_speed;
 	Asteroid_field.debris_genre = DG_DEBRIS;
 
-	for (int j = 0; j < MAX_ACTIVE_DEBRIS_TYPES; j++) {
-		Asteroid_field.field_debris_type[j] = -1;
-	}
+	Asteroid_field.field_debris_type.clear();
 
-	for (int j = 0; j < NUM_ASTEROID_SIZES; j++) {
-		Asteroid_field.field_asteroid_type[j] = false;
-	}
+	Asteroid_field.field_asteroid_type.clear();
 
-	int count = 0;
-	for (int i = 0; i < MAX_ACTIVE_DEBRIS_TYPES; i++) {
-		if (debris1 >= 0) {
-			Asteroid_field.field_debris_type[i] = debris1;
-			debris1 = -1;
-			count++;
-			continue;
-		}
-		if (debris2 >= 0) {
-			Asteroid_field.field_debris_type[i] = debris2;
-			debris2 = -1;
-			count++;
-			continue;
-		}
-		if (debris3 >= 0) {
-			Asteroid_field.field_debris_type[i] = debris3;
-			debris3 = -1;
-			count++;
-			continue;
-		}
-	}
+	Asteroid_field.field_debris_type = debris_types;
```

Comment:
Here, you copy debris_types twice. Once when you call the function, and then again here.
To avoid this, just replace this assignment here with a move assignment:
```suggestion
	Asteroid_field.field_debris_type = std::move(debris_types);
```